### PR TITLE
ingress interested

### PR DESCRIPTION
**Pull request by kortemik**

### CODE REVIEW

**Review by eemhu — APPROVED** [review]
LGTM

File: src/main/java/com/teragrep/net_01/channel/context/EstablishedContextImpl.java [hunk 1]
```diff
@@ -250,6 +243,12 @@ public Socket socket() {
         return socket;
     }
 
+    @Override
+    public Ingress ingress() {
+        // TODO this should trigger interestOps change for OP_READ but InterestOps is currently not thread safe
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/net_01/channel/context/EstablishedContextImpl.java:248]: Make an issue for this

File: src/main/java/com/teragrep/net_01/channel/context/IngressImpl.java [hunk 2]
```diff
@@ -250,4 +259,46 @@ private void activateBuffers(List<BufferLease> bufferLeases) {
     public AtomicBoolean needWrite() {
         return needWrite;
     }
+
+    @Override
+    public void register(Clock clock) {
+        lock.lock();
+        try {
+            if (!interestedClocks.isEmpty()) {
+                throw new IllegalStateException("only one interested clock is allowed");
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/net_01/channel/context/IngressImpl.java:268]: Maybe change the formatting to `Unable to register ingress clock, only one interested clock is allowed`

File: src/main/java/com/teragrep/net_01/channel/context/IngressImpl.java [hunk 3]
```diff
@@ -250,4 +259,46 @@ private void activateBuffers(List<BufferLease> bufferLeases) {
...
+            lock.unlock();
+        }
+    }
+
+    @Override
+    public void unregister(Clock clock) {
+        lock.lock();
+        try {
+            if (!interestedClocks.contains(clock)) {
+                throw new IllegalStateException("clock not registered");
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/net_01/channel/context/IngressImpl.java:283]: Maybe change the formatting to `Unable to unregister ingress clock, it is not registered`

**Review by StrongestNumber9 — CHANGES_REQUESTED** [review]
Minor changes, lgtm otherwise

**Review by StrongestNumber9 — APPROVED** [review]
lgtm

**Pull request merged**